controller/lib/core/sensors.cpp

```diff
@@ -56,6 +71,11 @@ void set_sensor_avg_samples(int numAvgSamples) {
 
 int get_sensor_avg_samples() { return sensorAvgSize; }
 
+static float diameterOfCircleToArea(float diameter) {
+  return static_cast<float>(M_PI *
```

Comment:
Nit, this is a little faster if we reorder it to M_PI / 4.0f * diameter * diameter.  Then the compiler can precompute M_PI / 4.